### ISSUE TITLE
java.lang.NoSuchMethodException: duke.main.Duke.<init>()

### ISSUE THREAD

**Issue by domlimm** [nus-cs2103-AY2122S2/forum#73]
Hello! Has anyone encountered this error?

I am on the last part of the JavaFX Tutorial. Can't quite figure out why this is happening.

I placed this into `Duke.java` instead of `Main.java`.

```java
@Override
    public void start(Stage stage) {
        try {
            FXMLLoader fxmlLoader = new FXMLLoader(Main.class.getResource("/view/MainWindow.fxml"));
            AnchorPane ap = fxmlLoader.load();
            Scene scene = new Scene(ap);
            stage.setScene(scene);
            fxmlLoader.<MainWindow>getController().setDuke(duke);
            stage.show();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
```

![image](https://user-images.githubusercontent.com/46486515/151915786-c9f15da7-c877-4ec8-9fa1-83a0a008cfd8.png)

I think that this error comes from the JavaFX portion, but i'm not sure from where...

Apologies for the lack of details, I know this isn't the best way to ask for help... Thought someone might have come across this when running Duke.

Here is the [link](https://github.com/domlimm/ip/tree/branch-Level-10/src/main) to `branch-Level-10` that might be helpful!

**Comment by RyanCheungJF** [comment]
Hi! I think this is because you don't have a constructor in Duke that doesn't take in any parameters, i.e `public Duke() { ... }` and if I'm not wrong that is what is required inside of `start`. I also tried this the other day and I referenced [this](https://stackoverflow.com/questions/49485351/javafx-exception-in-application-constructorno-such-method-found-exception).

**Comment by domlimm** [comment]
> Hi! I think this is because you don't have a constructor in Duke that doesn't take in any parameters, i.e `public Duke() { ... }` and if I'm not wrong that is what is required inside of `start`. I also tried this the other day and I referenced [this](https://stackoverflow.com/questions/49485351/javafx-exception-in-application-constructorno-such-method-found-exception).

Hello @RyanCheungJF ! I don't believe that is the issue... Java at compile time generates a default class constructor automatically. Thank you though!!!

**Comment by domlimm** [comment]
Alright, I fixed it by following the JavaFX Tutorial Part 4 instead of not creating another `Main` class to link `Duke` and `MainWindow`. Sorry about it @RyanCheungJF !

Also, with my [current implementation](https://github.com/domlimm/ip/tree/b715796c8501091f1e5de2c2f27b3d95ddb9e4fd), I managed to work around by ensuring that `Duke` has a constructor without any params.

```java
public Duke {
    // Inits
}
```

But I figured I should stick with the tutorial. Silly me!

Will close this now. Thanks!
